server/api.go

```diff
@@ -602,7 +607,7 @@ func (a *apiServer) subscribe(ctx context.Context, partition *partition,
 	)
 	if err != nil {
 		return nil, nil, status.New(
-			codes.Internal, fmt.Sprintf("Failed to create stream reader: %v", err))
+			codes.InvalidArgument, fmt.Sprintf("Failed to create stream reader: %v", err))
```

Comment:
I think you changed the code for the wrong error. `Internal` is correct here and line 600 should be an `InvalidArgument` IMO.

Reply:
ah, good spot